src/presentation/http/router/note.ts

```diff
@@ -264,15 +264,76 @@ const NoteRouter: FastifyPluginCallback<NoteRouterOptions> = (fastify, opts, don
   }, async (request, reply) => {
     const noteId = request.note?.id as number;
     const content = request.body.content as JSON;
-    const parentId = request.body.parentId;
 
-    const note = await noteService.updateNoteContentById(noteId, content, parentId);
+    const note = await noteService.updateNoteContentById(noteId, content);
 
     return reply.send({
       updatedAt: note.updatedAt,
     });
   });
 
+  /**
+   * Update note relation by id.
+   */
+  fastify.patch<{
+    Params: {
+      notePublicId: NotePublicId,
+    },
+    Body: {
+      parentNoteId: NotePublicId,
+    },
+    Reply: {
+      isUpdated: boolean,
+    }
+  }>('/:notePublicId/relation', {
+    schema: {
+      params: {
+        notePublicId: {
+          $ref: 'NoteSchema#/properties/id',
+        },
+      },
+      body: {
+        parentNoteId: {
+          $ref: 'NoteSchema#/properties/id',
+        },
+      },
+      response: {
+        '2xx': {
+          type: 'object',
+          properties: {
+            isUpdated: {
+              type: 'boolean',
+            },
+          },
+        },
+      },
+    },
+    config: {
+      policy: [
+        'authRequired',
+        'userCanEdit',
+      ],
+    },
+    preHandler: [
+      noteResolver,
+    ],
+  }, async (request, reply) => {
+    const noteId = request.note?.id as number;
+    const parentNoteId = request.body.parentNoteId;
+    let isUpdated: boolean | null = null;
+
+    try {
```

Comment:
Why not to place it in a domain level?

Reply:
Otherwise domain error won't be handled properly (it is why api was throwing 500 error).

Reply:
Maybe it should be fixed in our error handler?